### ISSUE TITLE
Should environment's name appear in KV name?

### ISSUE THREAD

**Issue by gregmsanderson** [cloudflare/wrangler#611]
#### Environment

* operating system:
Mac OS
* output of `rustc -V`:
* output of `node -v`:
v10.16.2
* output of `wrangler -V`:
👷 ✨  wrangler 1.3.1

#### Steps to reproduce

Mini issue that I don't know if this is a bug, or expected, or even which approach is better.

I tried the new environment commands to interact with KV via the CLI. Initially with 1.3.0 and then due a publish failure with that (due to a since-removed file as you speedily released 1.3.1) upgraded to 1.3.1. And I tried creating a KV for _dev_ and _prod_, the two environments I have set. Both worked, nice, but I noticed a slight change: _now_ when I make a prod one, 'prod' appears in the KV name. Previously it did not.

_Not_ having it in the _**KV** name_ seemed more consistent with the _**worker** name_ Since I'm using the approach of only declaring the name once at the top of wrangler.toml. And then letting wrangler append the env, as it desires. So it does for dev, and I assume it doesn't for prod. Haven't tried publishing a production one.

(I tried making a KV called 'config' initially and then one called 'kv')

**1.3.0**

$ wrangler kv:namespace create "config" --env prod
🌀  Creating namespace with title "worker-config"
✨  Success: WorkersKvNamespace {
    id: "...",
    title: "worker-config",
}
✨  Add the following to your wrangler.toml under [env.prod]:
kv-namespaces = [ 
	 { binding = "config", id = "..." } 
]
$ wrangler kv:namespace create "config" --env dev
🌀  Creating namespace with title "worker-dev-config"
✨  Success: WorkersKvNamespace {
    id: "...",
    title: "worker-dev-config",
}
✨  Add the following to your wrangler.toml under [env.dev]:
kv-namespaces = [ 
	 { binding = "config", id = "..." } 
]

**1.3.1**

$ wrangler kv:namespace create "kv" --env dev
🌀  Creating namespace with title "worker-dev-kv"
✨  Success: WorkersKvNamespace {
    id: "...",
    title: "worker-dev-kv",
}
✨  Add the following to your wrangler.toml under [env.dev]:
kv-namespaces = [ 
	 { binding = "kv", id = "..." } 
]
$ wrangler kv:namespace create "kv" --env prod
🌀  Creating namespace with title "worker-prod-kv"
✨  Success: WorkersKvNamespace {
    id: "...",
    title: "worker-prod-kv",
}
✨  Add the following to your wrangler.toml under [env.prod]:
kv-namespaces = [ 
	 { binding = "kv", id = "..." } 
]

#### What did you expect to see?

Both to match, either 'prod' appears in the KV's name for both, or neither

#### What did you see instead?

Now 'prod' appears in the KV name. Now, this isn't a big deal because the KV name is not seen of course, but I wondered if it would have implications if the code looked for a KV with 'prod', and doesn't find one, or vice versa. So in other words, code that works for 'dev' (where the behaviour is consistent) would break when published to 'prod'. Which wouldn't be good. Though perhaps it just uses the id internally and so the name doesn't matter.

**Comment by ashleymichal** [comment]
hey @gregmsanderson !

This is great feedback. The TL;DR is that ultimately, the namespace title is relatively inconsequential, _except_ that the API asserts that they be unique within a given account. Your Worker only references the namespace by the `binding` property, and Wrangler uses the `id` property to establish that binding for you.

So in order to keep uniqueness, but try to reinforce the relationship between the title, your worker script, and your binding, we structured the titles to be `<worker-name>-<binding>`

**Comment by gregmsanderson** [comment]
@ashleymichal Hmm ... I _think_ I had the [env.dev] and [env.prod] already set up in the toml before running the command. But now you ask, I'm only like 95% sure. So perhaps that was it.

I _then_ wondered if the order matters since I noticed I ran the prod and dev commands in reverse. Can't think why it would result in a different title system. I didn't try and duplicate that.

Out of interest I just deleted the prod KV, checked the toml had a [env.dev] and [env.prod] set, and re-ran the command to make it:

$ wrangler kv:namespace create "kv" --env prod
🌀  Creating namespace with title "worker-prod-kv"
✨  Success: WorkersKvNamespace {
    id: "...",
    title: "worker-prod-kv",
}
✨  Add the following to your wrangler.toml under [env.prod]:
kv-namespaces = [ 
	 { binding = "kv", id = "..." } 
]

... so it does continue to add 'prod' to the title of the KV. Which is fine and I'm reassured it doesn't matter since the id is the important bit in the binding. It sounds like the title is just for the humans.

If the KV name does or doesn't match the worker name (I guess 'worker-dev' for _dev_, and simply 'worker' for _prod_), well, no problem. It's clear which is the dev one.

Thanks.
